File: win-prog/ProgWinCS/Toolbars and Status Bars/SimpleStatusBar/SimpleStatusBar.cs
//----------------------------------------------
// SimpleStatusBar.cs © 2001 by Charles Petzold
//----------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class SimpleStatusBar: Form
{
     public static void Main()
     {
          Application.Run(new SimpleStatusBar());
     }
     public SimpleStatusBar()
     {
          Text = "Simple Status Bar";
		  ResizeRedraw = true;

               // Create status bar.

          StatusBar sb = new StatusBar();
          sb.Parent = this;
          sb.Text = "My initial status bar text";
     }
     protected override void OnPaint(PaintEventArgs pea)
     {
          Graphics grfx = pea.Graphics;
          Pen      pen  = new Pen(ForeColor);

          grfx.DrawLine(pen, 0, 0, ClientSize.Width, ClientSize.Height);
          grfx.DrawLine(pen, ClientSize.Width, 0, 0, ClientSize.Height);
     }
}

File: win-prog/ProgWinVB/Toolbars and Status Bars/SimpleStatusBar/SimpleStatusBar.vb
'------------------------------------------------
' SimpleStatusBar.vb (c) 2002 by Charles Petzold
'------------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class SimpleStatusBar
    Inherits Form

    Shared Sub Main()
        Application.Run(New SimpleStatusBar())
    End Sub

    Sub New()
        Text = "Simple Status Bar"
        ResizeRedraw = True

        ' Create status bar.
        Dim sbar As New StatusBar()
        sbar.Parent = Me
        sbar.Text = "My initial status bar text"
    End Sub

    Protected Overrides Sub OnPaint(ByVal pea As PaintEventArgs)
        Dim grfx As Graphics = pea.Graphics
        Dim pn As New Pen(ForeColor)

        grfx.DrawLine(pn, 0, 0, ClientSize.Width, ClientSize.Height)
        grfx.DrawLine(pn, ClientSize.Width, 0, 0, ClientSize.Height)
    End Sub
End Class
